Kernel/Syscalls/profiling.cpp

```diff
@@ -32,7 +32,6 @@ KResultOr<FlatPtr> Process::sys$profiling_enable(pid_t pid, u64 event_mask)
         else
             g_global_perf_events = PerformanceEventBuffer::try_create_with_size(32 * MiB).leak_ptr();
 
-        ScopedSpinLock lock(g_processes_lock);
         if (!TimeManagement::the().enable_profile_timer())
             return ENOTSUP;
         g_profiling_all_threads = true;
```

Comment:
What guards `g_profiling_all_threads`, etc... now that `g_processes_lock` is no longer held?

Reply:
A new `ContendedResource` or `ProtectedValue` ought to, since these variables by themselves aren't related to the global process list.

By the way, this shows that the old-school usage of synchronization primitives where the variables that need protection and the means to protect them being separate is far too easy to mess up, hence the point of this PR being to have better locking _tools_, not just better locking.

Reply:
> By the way, this shows that the old-school usage of synchronization primitives where the variables that need protection and the means to protect them being separate is far too easy to mess up, hence the point of this PR being to have better locking _tools_, not just better locking.

I understand, and agree... 😄 just pointing out that the current state has a few loose ends to tie up.